src/Application.php

```diff
@@ -218,12 +163,6 @@ public function finish($callback, $priority = 0)
                 2 => $this,
             ]);
 
-            if ($ret instanceof Response) {
-                $event->setResponse($ret);
-            } elseif (null !== $ret) {
-                throw new \RuntimeException('An after middleware returned an invalid response value. Must return null or an instance of Response.');
-            }
-
```

Comment:
It must have been overlooked when it was added. The [`TERMINATE`](https://github.com/symfony/symfony/blob/48710790afc1fb6e4c5cf42725cf6bf7a035225e/src/Symfony/Component/HttpKernel/KernelEvents.php#L99) kernel event is triggered after the response has been sent and thus doesn't allow to set a response. This change [aligns the code with Silex](https://github.com/silexphp/Silex/blob/master/src/Silex/Application.php#L298).